src/plugins/intel_gpu/src/runtime/layout.cpp

```diff
@@ -337,6 +337,32 @@ tensor layout::get_tensor() const {
     return t;
 }
 
+tensor layout::get_max_tensor() const {
+    if (!is_dynamic())
+        return get_tensor();
+    OPENVINO_ASSERT(
+        has_upper_bound(),
+        "[GPU] Dynamic layout called for get_max_tensor() should have upper bound (" + to_short_string() + ")");
+    ov::Shape shape;
+    for (auto dim : size) {
+        shape.push_back(dim.get_max_length());
+    }
+    std::vector<tensor::value_type> dims(shape.begin(), shape.end());
+
+    auto rank = std::max(format.dimension(), dims.size());
+    auto default_fmt = format::get_default_format(rank, format::is_weights_format(format), format::is_grouped(format));
+    if (default_fmt.dimension() > dims.size()) {
+        dims.insert(dims.end(), default_fmt.dimension() - dims.size(), 1);
+    }
+
+    while (dims.size() > default_fmt.dimension()) {
```

Comment:
Is it possible? As I understand, it occurs only if rank will be bigger than 6, and then maybe it's better to throw an exception instead of implicit reduce dimensions to bfyx format (which will be returned for rank size => 7)

Reply:
This part was just copied from get_tensor(). However now I think we dont need get_max_tensor separately for this PR's requirement . So I merged it to original get_max_tensor. Could you review again?